lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -2939,18 +2939,36 @@ ParseResult FIRStmtParser::parseRWProbe(Value &result) {
   // Not public port (verifier)
 
   // Check probe expression is base-type.
-  if (!type_isa<FIRRTLBaseType>(staticRef.getType()))
+  auto targetType = type_dyn_cast<FIRRTLBaseType>(staticRef.getType());
+  if (!targetType)
     return emitError(startTok.getLoc(),
                      "expected base-type expression in 'rwprobe', got ")
            << staticRef.getType();
 
-  // Check for other unsupported reference sources.
   auto fieldRef = getFieldRefFromValue(staticRef);
   auto target = fieldRef.getValue();
 
-  // TODO: Support for non-public ports.
-  if (isa<BlockArgument>(target))
-    return emitError(startTok.getLoc(), "rwprobe of port not yet supported");
+  // Ports are handled differently, emit a RWProbeOp with inner symbol.
+  if (auto arg = dyn_cast<BlockArgument>(target)) {
+    // Check target type.  Replicate inference/verification logic.
+    if (targetType.hasUninferredWidth() || targetType.hasUninferredReset())
+      return emitError(startTok.getLoc(),
+                       "must not have uninferred width or reset in type ")
+             << targetType;
```

Comment:
Nit: considering inverting the logic for readability: "must have known width or concrete reset type".